File: leetcode/001.Two_Sum.java
public class Solution {
    
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++) {
            if (map.get(nums[i]) != null) {
                return new int[]{map.get(nums[i])+1, i+1};
            }
            map.put(target - nums[i], i);
        }
        return new int[]{-1, -1};
    }
}

File: leetcode/001.Two_Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        map = {}
        for idx, val in enumerate(nums):
            if target-val not in map:
                map[val] = idx
            else:
                return [map[target-val]+1, idx+1]
        return [-1, -1]
